### PR TITLE
Double opt-in for outbox is no longer needed

### LINKED ISSUE #4183: Remove double opt-in for the Outbox
The Outbox currently requires a double opt-in mechanism via code & config file to be enabled. As raised in the discussion here https://github.com/Particular/NServiceBus/issues/3275 it doesn't seem to provide much value and may behave unexpectedly.

Is there any good reason why we should keep the double opt-in mechanism?

### PR DESCRIPTION
**Pull request by danielmarbach**
Closes #4183

### CODE REVIEW

**Comment by danielmarbach** [comment]
If we agree to continue I would update the upgrade guide

File: src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt.bak [hunk 1]
```diff
@@ -0,0 +1,3628 @@
...
+            where T : NServiceBus.Persistence.PersistenceDefinition { }
+        public static NServiceBus.PersistenceExtensions<T, S> UsePersistence<T, S>(this NServiceBus.EndpointConfiguration config)
+            where T : NServiceBus.Persistence.PersistenceDefinition
+            where S : NServiceBus.Persistence.StorageType { }
+        public static NServiceBus.PersistenceExtensions UsePersistence(this NServiceBus.EndpointConfiguration config, System.Type definitionType) { }
+    }
+    public class PersistenceExtensions : NServiceBus.Configuration.AdvanceExtensibility.ExposeSettings
+    {
+        public PersistenceExtensions(System.Type definitionType, NServiceBus.Settings.SettingsHolder settings, System.Type storageType) { }
+        [System.ObsoleteAttribute(@"Example: config.UsePersistence<InMemoryPersistence>().F
```
Comment (timbussmann) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt.bak]: what's this?

Comment (danielmarbach) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt.bak]: Slipped in. Will fix

**Comment by danielmarbach** [comment]
@timbussmann pushed

**Comment by andreasohlund** [comment]
>If we agree to continue I would update the upgrade guide

https://docs.particular.net/nservicebus/outbox/#enabling-the-outbox also needs to be changed

**Comment by timbussmann** [comment]
I'll pick that up since @danielmarbach is not around today/tomorrow.

**Comment by danielmarbach** [comment]
Thanks tim!

Tim Bussmann <notifications@github.com> schrieb am Mi. 25. Jan. 2017 um
12:27:

**Comment by timbussmann** [comment]
raised a pr against the outbox configuration page. Do we need to add something to the upgrade guide (as this is non breaking and we log a warning already)? @danielmarbach @SimonCropp

**Comment by SimonCropp** [comment]
re an upgrade guide. i dont think it is necessary, but if someone wants to go to the effort that is OK too

**Pull request merged**